### DISCUSSION TITLE
Git LFS with private JFrog Artifactory Repository

### DISCUSSION THREAD

**Discussion by rrpatil7279** [git-lfs/git-lfs#5549] [Q&A]
Hello There,

I have set up the Git LFS with JFrog Artifactory. I am keeping the large-sized files in Artifactory and the rest of the code in GitHub. Whenever I try to clone/push to my GitHub repo, Git asks me to enter credentials for JFrog Artifactory, and after that files are uploaded/downloaded from Artifcatory. Till this point, everything looks good!

I am setting the same thing with GitHub actions, but it's not working. Since my JFrog artifactory is private and does require credentials to be sent during code clone/push, I am looking for a way to programmatically send credentials. 

Any help on this matter would be highly appreciated. Thank You!

**Comment by bk2204** [comment]
Hey,

The easiest way to do this is with a credential helper, since Git LFS uses the same credential helpers as Git does.  If you have your Artifactory token in the `ARTIFACTORY_TOKEN` environment variable and your Artifactory server is on `https://example.com` with username `foo`, then you should be able to do this:

```
$ git config --global credential.helper store
$ printf 'url=https://foo@example.com\npassword=%s\n' "$ARTIFACTORY_TOKEN" | git credential approve
```

Then, once you've done that, the credential will be saved in a file on disk, and Git LFS will request it as normal when you're pushing or pulling.

The [Git FAQ also outlines how to use a shell script as a credential helper](https://git-scm.com/docs/gitfaq#http-credentials-environment), but I'd avoid that here because you're talking to two different servers and the example provided there doesn't account for different hostnames.  It's of course possible to adjust it using the documentation of the protocol in `git-credential(1)`, but it's tricky and this would be easier.
